qcodes/instrument_drivers/tektronix/AWG5014.py

```diff
@@ -1683,11 +1688,10 @@ def _pack_waveform(
             raise TypeError('Marker 2 contains invalid values.' +
                             ' Only 0 and 1 are allowed')
 
-        wflen = len(wf)
-        packed_wf = np.zeros(wflen, dtype=np.uint16)
-        packed_wf += np.uint16(np.round(wf * 8191) + 8191 +
-                               np.round(16384 * m1) +
-                               np.round(32768 * m2))
+        # Note: np.trunc is an order of magnitude faster then np.round
+        packed_wf = np.trunc(16384 * m1 + 32768 * m2
+                             + wf * 8191 + 8191.5).astype(np.uint16)
```

Comment:
is it correct that 8191 changed to 8191.5?

Reply:
It is 8191.5, because np.round has been replaced by np.trunc. For positive numbers this results in the same rounding with a very small exception. Python and numpy round half towards the nearest even value. So, 1.5 becomes 2 and 2.5 also becomes 2. That's what makes the Python and numpy implementations slow.
Added 0.5 and truncating rounds the half to the higher integer. So, 1.5 becomes 2 and 2.5 becomes 3.

Reply:
thank you for explaining!